code/modules/awaymissions/corpse.dm

```diff
@@ -4,294 +4,474 @@
+	var/id_access_list = null	//Allows you to manually add access to an ID card.
+	assignedrole = "Ghost Role"
+
+	var/husk = null
+	//these vars are for lazy mappers to override parts of the outfit
+	//these cannot be null by default, or mappers cannot set them to null if they want nothing in that slot
+	var/uniform = -1
+	var/r_hand = -1
+	var/l_hand = -1
+	var/suit = -1
+	var/shoes = -1
+	var/gloves = -1
+	var/ears = -1
+	var/glasses = -1
+	var/mask = -1
+	var/head = -1
+	var/belt = -1
+	var/r_pocket = -1
+	var/l_pocket = -1
+	var/back = -1
+	var/id = -1
+	var/neck = -1
+	var/backpack_contents = -1
+	var/suit_store = -1
+
+/obj/effect/mob_spawn/human/New()
+	if(ispath(outfit))
+		outfit = new outfit()
+	if(!outfit)
+		outfit = new /datum/outfit
+	return ..()
+
+/obj/effect/mob_spawn/human/equip(mob/living/carbon/human/H)
 	if(mob_species)
-		M.set_species(mob_species)
-	if(corpseuniform)
-		M.equip_to_slot_or_del(new corpseuniform(M), slot_w_uniform)
-	if(corpsesuit)
-		M.equip_to_slot_or_del(new corpsesuit(M), slot_wear_suit)
-	if(corpseshoes)
-		M.equip_to_slot_or_del(new corpseshoes(M), slot_shoes)
-	if(corpsegloves)
-		M.equip_to_slot_or_del(new corpsegloves(M), slot_gloves)
-	if(corpseradio)
-		M.equip_to_slot_or_del(new corpseradio(M), slot_l_ear)
-	if(corpseglasses)
-		M.equip_to_slot_or_del(new corpseglasses(M), slot_glasses)
-	if(corpsemask)
-		M.equip_to_slot_or_del(new corpsemask(M), slot_wear_mask)
-	if(corpsehelmet)
-		M.equip_to_slot_or_del(new corpsehelmet(M), slot_head)
-	if(corpsebelt)
-		M.equip_to_slot_or_del(new corpsebelt(M), slot_belt)
-	if(corpsepocket1)
-		M.equip_to_slot_or_del(new corpsepocket1(M), slot_r_store)
-	if(corpsepocket2)
-		M.equip_to_slot_or_del(new corpsepocket2(M), slot_l_store)
-	if(corpseback)
-		M.equip_to_slot_or_del(new corpseback(M), slot_back)
-	if(corpseid == 1)
-		var/obj/item/weapon/card/id/W = new(M)
-		W.name = "[M.real_name]'s ID Card"
-		var/datum/job/jobdatum
-		for(var/jobtype in typesof(/datum/job))
-			var/datum/job/J = new jobtype
-			if(J.title == corpseidaccess)
-				jobdatum = J
-				break
-		if(corpseidicon)
-			W.icon_state = corpseidicon
-		if(corpseidaccess)
-			if(jobdatum)
-				W.access = jobdatum.get_access()
-			else
+		H.set_species(mob_species)
+	if(husk)
+		H.ChangeToHusk()
+	else //Because for some reason I can't track down, things are getting turned into husks even if husk = false. It's in some damage proc somewhere.
+		H.mutations.Remove(HUSK)
+	H.underwear = "Nude"
+	H.undershirt = "Nude"
+	H.socks = "Nude"
+	if(outfit)
+		var/static/list/slots = list("uniform", "r_hand", "l_hand", "suit", "shoes", "gloves", "ears", "glasses", "mask", "head", "belt", "r_pocket", "l_pocket", "back", "id", "neck", "backpack_contents", "suit_store")
+		for(var/slot in slots)
+			var/T = vars[slot]
+			if(!isnum(T))
+				outfit.vars[slot] = T
+		H.equipOutfit(outfit)
+		//if(disable_pda) //FETHAS FIX THIS

```

Comment:
Commented code.

Reply:
its commeted out cuase i am not sure how to disable a PDA properly.

Reply:
Maybe just don't give them PDAs?

Reply:
well that won;t work with the AI.

Reply:
Well, something should be done.... its probably not good to add stuff with a glaring FETHAS FIX THIS next to it.

Reply:
i know its more a note to gifure out what to do or get a idea on how to fix